app/code/Magento/Braintree/Test/Unit/Gateway/Response/VaultDetailsHandlerTest.php

```diff
@@ -72,12 +69,11 @@ class VaultDetailsHandlerTest extends \PHPUnit\Framework\TestCase
     protected function setUp()
     {
         $this->paymentToken = $this->createMock(PaymentTokenInterface::class);
-        $this->paymentTokenFactory = $this->getMockBuilder(CreditCardTokenFactory::class)
+        $this->paymentTokenFactory = $this->getMockBuilder(PaymentTokenFactoryInterface::class)
             ->setMethods(['create'])
             ->disableOriginalConstructor()
             ->getMock();
-        $this->paymentTokenFactory->expects(self::once())
-            ->method('create')
+        $this->paymentTokenFactory->method('create')
```

Comment:
1. Agree.
2. I don't see how it will make the test more useful, it doesn't matter how many times the `create` method will be called, the main purpose of this factory - create the instance of the class.

Reply:
Previously it was checking exactly how much times it will be called, so it was testing business logic. After your changes - it will not fail in case when it will be executed few times (that was not expected before)

Reply:
>  it was testing business logic
Actually, it just checking if the method called but not how business logic works. I think it better to add additional `asserts` to check if payment token will be created with all needed data rather then just check if the method called or not.